bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/handler/RoombaHandler.java

```diff
@@ -189,9 +189,38 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                         String[] params = cmds[1].split(";");
 
                         String mapId = params[0];
-                        String[] regionIds = params[1].split(",");
-
-                        MQTTProtocol.Request request = new MQTTProtocol.CleanRoomsRequest("start", mapId, regionIds);
+                        String user_mapv_id;
+                        if (params.length >= 3) {
+                            user_mapv_id = params[2];
+                        } else {
+                            user_mapv_id = null;
+                        }
+
+                        String[] ArrayRegions = params[1].split(",");
+                        String regionIds[] = new String[ArrayRegions.length];
+                        String regionType[] = new String[ArrayRegions.length];
+
+                        for (int i = 0; i < ArrayRegions.length; i++) {
+                            String[] ArrayRegionDetails = ArrayRegions[i].split("=");
```

Comment:
Naming conventions: `arrayRegionDetails` instead of `ArrayRegionDetails`.
`regionDetails` would even be better.